### ISSUE TITLE
Look through ruby/reek

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#422]
Are there any new rules that should be ported to our linter?
https://github.com/troessner/reek#code-smells

**Comment by sobolevn** [comment]
Things I like:

- [ ] https://github.com/troessner/reek/blob/master/docs/Control-Parameter.md

**Comment by proofit404** [comment]
`yield` in ruby is something similar to generator `next` or `send` methods in python. But it working with ruby blocks instead of python generators.

The closest example for python will be something like this:

```python
def api_func(callback):
    value = get_value()
    callback.send(value) # `yield` will be here in ruby.
    return value

def callback_gen():
    value = yield # |value| block argument will be here in ruby
    process(value)

callback = callback_gen()
next(callback) # init generator

api_func(callback) # <callback_gen> `end` will be here in `ruby`.
```

**Comment by sobolevn** [comment]
Thanks for the clarification! 

Still, returning or yielding to many parameters is not a good thing. And we can cover it with linter (with some limitations).
I will create a rule request for this.

**Comment by proofit404** [comment]
I think we can also add "too many `await` statements" rule.

**Comment by sobolevn** [comment]
Yes, we should also cover cases like `await one, two, three, four`
It should share the same semantics as `return` and `yield`. Good catch!
